### PR TITLE
Block Component

### PR DESCRIPTION
**Pull request by kaikun213**
Component to display a block with optional details fast and easy.
Missing: Nonce, BlockReward, UnclesReward. Formatting of BigNumber difficulty.
Any other defaults?

### CODE REVIEW

File: src/Block.jsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                      <List.Content floated='right'>
+                        <div style={{fontSize:'x-small'}}>{this.state.block.timestamp.toString()}</div>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {this.props.transactions ?  <List.Item>
+                      <List.Content>
+                        Transactions
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Ra>{this.state.block.transactions.length}</Ra> txns
```
Comment (gavofyork) [inline src/Block.jsx]: `<Ra>` is the Bond-happy version of `<a>` - I think you just mean `<Rspan>`...

File: src/Block.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                      <List.Content>
+                        Sha3Uncles
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Hash value={this.state.block.sha3Uncles}></Hash>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {(this.props.author || this.props.miner) ?  <List.Item>
+                      <List.Content floated='right'>
+                        <Hash value={this.state.block.author}></Hash>
```
Comment (gavofyork) [inline src/Block.jsx]: Can use `<InlineAccount>` instead.

File: src/Block.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                        <Rspan style={{fontSize:'small'}}>{this.state.block.gasUsed.toString(10)}</Rspan>
+                      </List.Content>
+                    </List.Item> : "" }
+                    {/* TODO: BlockReward, UncleReward, Nonce */}
+                    {this.props.extraData ?  <List.Item>
+                      <List.Content>
+                        Extra Data
+                      </List.Content>
+                      <List.Content floated='right'>
+                        <Hash value={this.state.block.extraData}></Hash>
```
Comment (gavofyork) [inline src/Block.jsx]: Should probably be formatted as a straight hex string and ASCII representation.

File: src/Transaction.jsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+                          <List.Content floated='right'>
+                            <Rspan>{formatBlockNumber(this.state.transaction.blockNumber)}</Rspan>
+                          </List.Content>
+                        </List.Item> : "" }
+                        {this.props.from ?  <List.Item>
+                          <List.Content>
+                            From
+                          </List.Content>
+                          <List.Content floated='right'>
+                            <Hash value={this.state.transaction.from}></Hash>
```
Comment (gavofyork) [inline src/Transaction.jsx]: `<InlineAccount>`

File: src/Transaction.jsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+                          </List.Content>
+                        </List.Item> : "" }
+                        {this.props.to ?  <List.Item>
+                          <List.Content>
+                            To
+                          </List.Content>
+                          <List.Content floated='right'>
+                            <Hash value={this.state.transaction.to}></Hash>
```
Comment (gavofyork) [inline src/Transaction.jsx]: `<InlineAccount>`

File: src/Transaction.jsx [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+                          </List.Content>
+                        </List.Item> : "" }
+                        {this.props.ether ?  <List.Item>
+                          <List.Content>
+                            Ether
+                          </List.Content>
+                          <List.Content floated='right'>
+                            <Rspan>{formatBalance(this.state.transaction.value)}</Rspan>
```
Comment (gavofyork) [inline src/Transaction.jsx]: `<InlineBalance>`

File: src/Transaction.jsx [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+                          <List.Content floated='right'>
+                            <Rspan>{formatCondition(this.state.transaction.condition)}</Rspan>
+                          </List.Content>
+                        </List.Item> : "" }
+                        {this.props.creates ? <List.Item>
+                          <List.Content>
+                            Creates
+                          </List.Content>
+                          <List.Content floated='right'>
+                            <Hash value={this.state.transaction.creates}></Hash>
```
Comment (gavofyork) [inline src/Transaction.jsx]: `<InlineAccount>`

File: src/Transaction.jsx [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+                          <List.Content floated='right'>
+                            <Rspan>{this.state.transaction.gas.toString(10)}</Rspan>
+                          </List.Content>
+                        </List.Item> : "" }
+                        {this.props.gasPrice ? <List.Item>
+                          <List.Content>
+                            Gas Price
+                          </List.Content>
+                          <List.Content floated='right'>
+                            <Rspan>{formatBalance(this.state.transaction.gasPrice)}</Rspan>
```
Comment (gavofyork) [inline src/Transaction.jsx]: `<InlineBalance>`

File: src/Transaction.jsx [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+                          </List.Content>
+                        </List.Item> : "" }
+                        {this.props.input ? <List.Item>
+                          <List.Content>
+                            Input Data
+                          </List.Content>
+                          <List.Content floated='right'>
+                            {/* Input is raw byte data */}
+                            <Hash value={this.state.transaction.input}></Hash>
```
Comment (gavofyork) [inline src/Transaction.jsx]: Better to format as raw bytes; ideally it would eventually parse ABIs, but that's definitely for version 2 :-)

**Review by gavofyork — CHANGES_REQUESTED** [review]
A few missed opportunities to use existing components there. Looks good otherwise.

File: src/Block.jsx (hunk 2 above)
Comment (kaikun213) [reply src/Block.jsx]: Didn't want to double the icon but yes looks better. Will see if the other fields can look as nice too.

**Pull request merged**